src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_SolarTower.java

```diff
@@ -551,45 +556,51 @@ private boolean addSolarHeater(IGregTechTileEntity aTileEntity, int a) {
 
 	@Override
 	public boolean checkRecipe(final ItemStack aStack) {
-		this.mEfficiencyIncrease = 10;
-		this.mMaxProgresstime = 100;
+		this.mEfficiencyIncrease = 100;
+		this.mMaxProgresstime = 200;
 
 		if (this.mSolarHeaters.isEmpty() || this.mSolarHeaters.size() < 340 || this.getTotalRuntimeInTicks() % 200 == 0) {
 			getConnectedSolarReflectors();			
 		}
 
 		int aTier = getHeaterTier();
 		int aHeaters = getHeaterCountForTier(aTier);
-		
-		//Manage Heat every 5s
-		//Add Heat First, if sources available
-		if (aHeaters > 0) {
-			for (int i = 0; i < aHeaters; i++) {
-				Math.min((this.mHeatLevel += aTier), 20000);
-			}
+
+		double aEfficiency;
+		// Original formula was (-Math.pow(this.mHeatLevel - 50000, 0.8) + 7000) / 7000
+		// However, negative numbers to the power of a non-integer result in NaN, by default
+		// Max efficiency is 1, at mHeatLevel = 50000, and it lowers at the same rate if going above or below this heat
+		// Min efficiency is 0.179, at mHeatLevel = 0 or 100000
+		if (this.mHeatLevel >= 50000) {
+			aEfficiency = (-Math.pow(this.mHeatLevel - 50000, 0.8) + 7000) / 7000;
 		}
-		
-		//Remove Heat, based on time of day
-		if (mHeatLevel > 0) {
-			if (mHeatLevel > 20000) {
-				this.mHeatLevel = 20000;
-			}
-		}		
+		else {
+			aEfficiency = (-Math.pow((100000 - this.mHeatLevel) - 50000, 0.8) + 7000) / 7000;
+		}
+
 		World w = this.getBaseMetaTileEntity().getWorld();
+
+		//Manage Heat every 5s
+		//Add Heat First, if sources available and it's daytime
 		if (w != null) {
-			int aRemovalFactor = 0;
-			if (w.isDaytime()) {
-				aRemovalFactor = 1;
+			if (aHeaters > 0 && w.isDaytime()) {
```

Comment:
Check also if it's raining. This is taken from solar panel from GT:
```
aTileEntity.getWorld().isRaining() && aTileEntity.getBiome().rainfall > 0.0F
```

Reply:
I completely forgot about rain. I'm not sure if it's really important for this multi, but in that case, which tile entity checks for rain? The reflectors are all tile entities, and this is a multi-chunk multiblock, unlike the GT solars.

Reply:
Reflectors already check `canBlockSeeTheSky`, so rain check can be done only with controller

Reply:
Done, I think I did it right.